src/arch/arm/kernel/boot.c

```diff
@@ -364,6 +360,12 @@ static BOOT_CODE bool_t try_init_kernel(
     /* initialise the platform */
     init_plat();
 
+    if (it_v_reg.end >= USER_TOP) {
+        printf("ERROR: userland image virt [%p..%p] exceeds USER_TOP (%p)\n",
```

Comment:
What's the rationale for casting to `(void *)` and using `%p`? Would it be clearer to simply use `%lx` ?

Reply:
We can't use `%lx` but it would have to be `SEL4_PRIx_word` then to do it right. At some point I just got a bit lazy to do it everywhere if this is not absolutely necessary - like here in case of virtual addresses and a `prinf()`. If we are dealing with a virtual address that is in the kernel's address space, this is semantically something we can cast to `void*` safely. Also, that saves a bit of typing and make it  a bit nicer to read. Besides `%p` was invented to print pointers and most compilers/analyzer are aware of that that concept, so I'm just using this.

Reply:
It isn't invalid for it_v_reg.start to be 0 in which case %p will format it to "(nil)" compared to 0x%lx which will format it as 0x0.

Reply:
@kent-mcleod: good point, even if our `printf()` just prionts `0` here.
I've changes this to use `SEL4_PRIx_word` now. Seem the definition of the constant `USER_TOP` could get some cleanup across all platforms to either make it `ul` or `void*` or `word_t` everywhere. A lot platforms use `ul`, is this the type to use for integers that are pointers? It's a pity we don't have the generic `unitptr_t` for this.

Reply:
History lesson: For a long time `%p` was not implemented in the kernel's printf, which is why you will see `0x%lx` everywhere.